cin_validator/rules/cin2022_23/rule_8750.py

```diff
@@ -0,0 +1,166 @@
+                "GenderCurrent": "0",
+            },
+            {  # 3 - Pass - Not unborn or born! (Not relevant to this rule)
+                "LAchildID": "child3",
+                "PersonBirthDate": pd.NA,
+                "ExpectedPersonBirthDate": pd.NA,
+                "GenderCurrent": "2",
+            },
+            {  # 4 - Fail - Unborn with Gender = 2
+                "LAchildID": "child4",
+                "PersonBirthDate": pd.NA,
+                "ExpectedPersonBirthDate": "25/05/2000",
+                "GenderCurrent": "2",
+            },
+        ]
+    )
+    # if rule requires columns containing date values, convert those columns to datetime objects first. Do it here in the test_validate function, not above.
+    child_identifiers[PersonBirthDate] = pd.to_datetime(
+        child_identifiers[PersonBirthDate], format="%d/%m/%Y", errors="coerce"
+    )
+    child_identifiers[ExpectedPersonBirthDate] = pd.to_datetime(
+        child_identifiers[ExpectedPersonBirthDate], format="%d/%m/%Y", errors="coerce"
+    )
+
+    # Run rule function passing in our sample data
+    result = run_rule(validate, {ChildIdentifiers: child_identifiers})
+
+    # Use .type1_issues to check for the result of .push_type1_issues() which you used above.
+    issues = result.type1_issues
+
+    # get table name and check it. Replace ChildProtectionPlans with the name of your table.
+    issue_table = issues.table
+    assert issue_table == ChildIdentifiers
+
+    # check that the right columns were returned. Replace CPPstartDate and CPPendDate with a list of your columns.
+    issue_columns = issues.columns
+    assert issue_columns == [GenderCurrent, PersonBirthDate]
+
+    # check that the location linking dataframe was formed properly.
+    issue_rows = issues.row_df
+    # replace 2 with the number of failing points you expect from the sample data.
+    assert len(issue_rows) == 1
+    # check that the failing locations are contained in a DataFrame having the appropriate columns. These lines do not change.
+    assert isinstance(issue_rows, pd.DataFrame)
+    assert issue_rows.columns.to_list() == ["ERROR_ID", "ROW_ID"]
+
+    # Create the dataframe which you expect, based on the fake data you created. It should have two columns.
+    # - The first column is ERROR_ID which contains the unique combination that identifies each error instance, which you decided on earlier.
+    # - The second column in ROW_ID which contains a list of index positions that belong to each error instance.
+
+    # The ROW ID values represent the index positions where you expect the sample data to fail the validation check.
+    expected_df = pd.DataFrame(
+        [
+            {
+                "ERROR_ID": (
+                    "child4",
+                    "2",
+                    pd.NaT,
+                ),
+                "ROW_ID": [4],
+            },
+        ]
+    )
+    assert issue_rows.equals(expected_df)
```

Comment:
3/3
expected_df needs to be changed to reflect the updated ERROR_ID combination.